### PR TITLE
[Feature] Add feature loss for DICGAN

### PR DESCRIPTION
**Pull request by Yshuo-Li**

### CODE REVIEW

File: mmedit/models/losses/feature_loss.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    It is used to train DICGAN.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        model = LightCNN(3)
+        self.features = nn.Sequential(*list(model.features.children()))
+        for _, v in self.features.named_parameters():
```
Comment (ckkelvinchan) [inline mmedit/models/losses/feature_loss.py:21]: `self.features.requires_grad_(False)`

File: mmedit/models/losses/feature_loss.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        Args:
+            x (Tensor): Input tensor.
+
+        Returns:
+            Tensor: Forward results.
+        """
+
+        return self.features(x)
+
+    def init_weights(self, pretrained=None, strict=True):
```
Comment (ckkelvinchan) [inline mmedit/models/losses/feature_loss.py:36]: Will you call this function?

File: mmedit/models/losses/feature_loss.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        if isinstance(pretrained, str):
+            logger = get_root_logger()
+            load_checkpoint(self, pretrained, strict=strict, logger=logger)
+        elif pretrained is not None:
+            raise TypeError(f'"pretrained" must be a str or None. '
+                            f'But received {type(pretrained)}.')
+
+
+@LOSSES.register_module()
+class FeatureLoss(nn.Module):
```
Comment (ckkelvinchan) [inline mmedit/models/losses/feature_loss.py:54]: I think this naming is a bit too general. Since it is specfic to the LightCNN, maybe change to a more specific name?

Or maybe this loss can be moved to model, by computing the feature and directly use L1Loss?

File: mmedit/models/losses/feature_loss.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        assert isinstance(pretrained, str), 'Model must be pretrained'
+        self.model.init_weights(pretrained)
+        self.model.eval()
+        self.loss_weight = loss_weight
+        if criterion == 'l1':
+            self.criterion = torch.nn.L1Loss()
+        elif criterion == 'mse':
+            self.criterion = torch.nn.MSELoss()
+        else:
+            raise ValueError("'criterion' should be 'l1' or 'mse',"
```
Comment (ckkelvinchan) [inline mmedit/models/losses/feature_loss.py:76]: Add a space after `,`.

File: tests/test_models/test_losses/test_feature_loss.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import pytest
+import torch
+
+from mmedit.models.losses import FeatureLoss
+
+
+def test_feature_loss():
+    pretrained = 'https://download.openmmlab.com/mmediting/' + \
```
Comment (ckkelvinchan) [inline tests/test_models/test_losses/test_feature_loss.py:8]: There may be a problem loading checkpoints during test. Need confirm with @innerlee

File: tests/test_models/test_losses/test_feature_loss.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    # test criterion value error
+    with pytest.raises(ValueError):
+        FeatureLoss(pretrained=pretrained, criterion='l2')
+    # test assert isinstance(pretrained, str)
+    with pytest.raises(AssertionError):
+        FeatureLoss(pretrained=None)
+
+
+if __name__ == '__main__':
```
Comment (ckkelvinchan) [inline tests/test_models/test_losses/test_feature_loss.py:51]: Remove this part.

File: mmedit/models/losses/feature_loss.py (hunk 2 above)
Comment (Yshuo-Li) [reply mmedit/models/losses/feature_loss.py:36]: Yes, the model should be pre-trained in training and testing.

**Review by ckkelvinchan — APPROVED** [review]
No problem except the doubt about model loading in unittest.

File: mmedit/models/losses/feature_loss.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        criterion (str): Criterion type. Options are 'l1' and 'mse'.
+            Default: 'l1'.
+    """
+
+    def __init__(self, pretrained, loss_weight=1.0, criterion='l1'):
+        super().__init__()
+        self.model = LightCNNFeature()
+        assert isinstance(pretrained, str), 'Model must be pretrained'
+        self.model.init_weights(pretrained)
+        self.model.eval()
```
Comment (innerlee) [inline mmedit/models/losses/feature_loss.py:68]: Should this "eval" status be kept during training? If so, how about add an assertion at L88 in the forward process

**Review by innerlee — APPROVED** [review]
